src/main/kotlin/com/terraformation/backend/documentproducer/db/VariableStore.kt

```diff
@@ -85,7 +93,23 @@ class VariableStore(
    *   children.
    */
   fun fetchVariable(manifestId: VariableManifestId, variableId: VariableId): Variable {
-    return variables[manifestId to variableId] ?: FetchContext(manifestId).fetchVariable(variableId)
+    return variables[manifestId to variableId]
+        ?: ManifestFetchContext(manifestId).fetchVariable(variableId)
+  }
+
+  /**
+   * Returns information about a variable. Eagerly fetches other variables that are related to the
+   * requested variable, e.g., table columns.
+   *
+   * @throws VariableNotFoundException The variable didn't exist.
+   * @throws VariableIncompleteException The variable didn't have all the information required for
+   *   its type.
+   * @throws CircularReferenceException There is a cycle in the graph of the variable and its
+   *   children.
+   */
+  fun fetchVariableForStableId(stableId: String, variableId: VariableId): Variable {
```

Comment:
Why are both the stable and variable IDs needed here? Can't you look up the stable ID given just the variable ID?

Reply:
That is true, I was trying to repurpose the code used for the manifest import, but it seems like it is unnecessary as you point out